helpers/template.go

```diff
@@ -128,36 +128,31 @@ func NewSecretTransformationOption(ctx context.Context, client ctrlclient.Client
 		return nil, err
 	}
 
-	transformation := meta.GetTransformation()
-	if transformation == nil {
-		return nil, nil
-	}
-
-	return NewSecretTransformationOptionWithTransformation(ctx, client, obj, transformation, globalOpt)
+	return NewSecretTransformationOptionWithTransformation(ctx, client, obj, meta.GetTransformation(), globalOpt)
 }
 
 func NewSecretTransformationOptionWithTransformation(ctx context.Context, client ctrlclient.Client, obj ctrlclient.Object, transformation *secretsv1beta1.Transformation,
 	globalOpt *GlobalTransformationOptions,
 ) (*SecretTransformationOption, error) {
-	keyedTemplates, ff, err := gatherTemplates(ctx, client, obj, transformation)
-	if err != nil {
-		return nil, err
-	}
+	opt := &SecretTransformationOption{}
+	if transformation != nil {
+		keyedTemplates, ff, err := gatherTemplates(ctx, client, obj, transformation)
```

Comment:
What is the difference in behavior now that gatherTemplates is taking an empty `opt` object, instead of the one passed in by the function?